### PR TITLE
[BEAM-6472] Set read permissions for the staged files

### PR DESCRIPTION
**Pull request by angoenka**
Read permission are necessary for non root user.

**Please** add a meaningful description for your change here

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by angoenka** [comment]
cc: @mxm @robertwb

File: sdks/python/apache_beam/runners/portability/portable_stager.py [hunk 1]
```diff
@@ -71,7 +71,8 @@ def stage_artifact(self, local_path_to_artifact, artifact_name):
     def artifact_request_generator():
       artifact_metadata = beam_artifact_api_pb2.ArtifactMetadata(
           name=artifact_name,
-          sha256=_get_file_hash(local_path_to_artifact))
+          sha256=_get_file_hash(local_path_to_artifact),
+          permissions=444)
```
[the remark this answers is not in the document]
Comment (angoenka) [reply sdks/python/apache_beam/runners/portability/portable_stager.py]: Artifact retrieval service download and create artifact on sdk harness and sets the file permission here https://github.com/apache/beam/blob/7b9fd010886f2bb2bc0cec552666b1b35a59324f/sdks/go/pkg/beam/artifact/materialize.go#L153

As we don't pass any permission, no permissions are set on file creation at the time of retrieval.
If SDK Harness is not executed as root which is the most common case in non dokorized environment, artifacts can not be read.

[the remark this answers is not in the document]
Comment (mxm) [reply sdks/python/apache_beam/runners/portability/portable_stager.py]: I had assumed no permissions would mean default permissions, i.e. read and write access but not executable. Thanks!

[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/runners/portability/portable_stager.py]: Does this make files staged to, e.g., GCS publicly readable?

[the remark this answers is not in the document]
Comment (angoenka) [reply sdks/python/apache_beam/runners/portability/portable_stager.py]: No, they should not. 
The permissions are stored as metadata and are applied on retrieval.

**Pull request merged**